contracts/src/TeleporterTokenDestination.sol

```diff
@@ -541,6 +346,252 @@ abstract contract TeleporterTokenDestination is
         return (amount, primaryFee);
     }
 
+    /**
+     * @dev Sends tokens to the specified destination token bridge instance.
+     *
+     * Requirements:
+     *
+     * - {input.destinationBridgeAddress} cannot be the zero address
+     * - {input.recipient} cannot be the zero address
+     * - {amount} must be greater than 0
+     */
+    function _processSend(SendTokensInput calldata input, uint256 amount) private {
+        // If the destination blockchain is the source bridge instance's blockchain,
+        // the destination bridge address must match the token source address,
+        // and no secondary fee or fallback is needed.
+        require(
+            input.destinationBridgeAddress == tokenSourceAddress,
+            "TeleporterTokenDestination: invalid destination bridge address"
+        );
+        require(input.secondaryFee == 0, "TeleporterTokenDestination: non-zero secondary fee");
+        require(
+            input.multiHopFallback == address(0),
+            "TeleporterTokenDestination: non-zero multi-hop fallback"
+        );
```

Comment:
Should we move these checks to a `_validateSingleHopInput` helper? It looks like the same set of checks are duplicated in `_processSendAndCall`, so would great to de-duplicate.

Likely applies to the checks in `_processSendMultiHop` and `_processSendAndCallMultiHop` also

Reply:
Yeah, unfortunately each of the functions uses a different input data type, so we can't pass it in whole, but I made the helper functions with the individual parameters, which still simplifies the code.